exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricExporter.java

```diff
@@ -0,0 +1,205 @@
+package com.google.cloud.opentelemetry.metric;
+
+import static com.google.api.client.util.Preconditions.checkNotNull;
+import static com.google.cloud.opentelemetry.metric.MetricTranslator.mapGcpMonitoredResource;
+import static com.google.cloud.opentelemetry.metric.MetricTranslator.mapMetric;
+import static com.google.cloud.opentelemetry.metric.MetricTranslator.mapMetricDescriptor;
+import static com.google.cloud.opentelemetry.metric.MetricTranslator.mapPoint;
+import static java.util.logging.Level.WARNING;
+
+import com.google.api.Metric;
+import com.google.api.MetricDescriptor;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.monitoring.v3.MetricServiceClient;
+import com.google.cloud.monitoring.v3.MetricServiceSettings;
+import com.google.cloud.monitoring.v3.stub.MetricServiceStub;
+import com.google.common.collect.Lists;
+import com.google.monitoring.v3.CreateMetricDescriptorRequest;
+import com.google.monitoring.v3.Point;
+import com.google.monitoring.v3.ProjectName;
+import com.google.monitoring.v3.TimeSeries;
+import io.opentelemetry.common.Labels;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.logging.Logger;
+
+public class MetricExporter implements io.opentelemetry.sdk.metrics.export.MetricExporter {
+
+  private static final String PROJECT_NAME_PREFIX = "projects/";
+  private static final long WRITE_INTERVAL_SECOND = 10;
+  private static final int MAX_BATCH_SIZE = 200;
+  private static final long NANO_PER_SECOND = (long) 1e9;
+
+  private static final Logger logger = Logger.getLogger(MetricExporter.class.getName());
+
+  private final MetricServiceClient metricServiceClient;
+  private final String projectId;
+  private final Instant exporterStartTime;
+  private final Map<MetricWithLabels, Long> lastUpdatedTime = new HashMap<>();
+  private String uniqueIdentifier;
+
+  MetricExporter(
+      String projectId,
+      MetricServiceClient client,
+      boolean addUniqueIdentifier
+  ) {
+    this.projectId = projectId;
+    this.metricServiceClient = client;
+    if (addUniqueIdentifier) {
+      this.uniqueIdentifier = UUID.randomUUID().toString().replaceAll("-", "").substring(0, 8);
```

Comment:
Might be worth checking with @aabmass as I can't remember all the details.

But I recall the main reason for this was to avoid having errors on the Stackdriver side if you send the same metric from two different applications (since Stackdriver doesn't have any equivalent of collection aggregation and gives an error if it receives the same metric timeseries info within 10s). This should not be an issue in most cases as metric data will be sent from different **resources** (this doesn't cover the case of having two processes on the same machine, but for that edge case I think it's reasonable for the user to handle that themselves).

==> I would recommend not implementing this unless we find a good use case for it.

Reply:
I agree. @zoercai could you please just remove all usage of "unique identifier" for now?